crates/biome_js_analyze/src/analyzers/nursery/no_done_callback.rs

```diff
@@ -0,0 +1,168 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::{
+    AnyJsArrowFunctionParameters, AnyJsBinding, AnyJsExpression, JsCallExpression, JsParameters,
+    JsTemplateExpression,
+};
+use biome_rowan::{AstNode, TextRange};
+
+declare_rule! {
+    /// This rule checks the function parameter of hooks & tests for use of the done argument, suggesting you return a promise instead.
```

Comment:
Following existing practices, it would be nice to add a sentence describing what this is.
```suggestion
    /// Disallow using a callback in asynchronous tests and hooks.
    ///
    /// This rule checks the function parameter of hooks & tests for use of the done argument, suggesting you return a promise instead.
```